File: web/capa5/types/meeting.ts
import { DateTime } from 'luxon';
import { Uniform } from '~/types/uniforms';

export const enum MeetingType {
    RegularMeeting = 'regular meeting',
    SpecialActivity = 'special activity',
}

export class Meeting {
    location: string;
    address: string;
    start: DateTime;
    end: DateTime;
    meetingType: MeetingType;
    topic: string;
    blocks: MeetingBlock[];
    uod: Uniform[];
    additionalOrders: string;

    constructor(
        location: string,
        address: string,
        start: DateTime,
        end: DateTime,
        meetingType: MeetingType,
        topic: string,
        blocks: MeetingBlock[],
        uod: Uniform[],
        additionalOrders: string,
    ) {
        this.location = location;
        this.address = address;
        this.start = start;
        this.end = end;
        this.meetingType = meetingType;
        this.topic = topic;
        this.blocks = blocks;
        this.uod = uod;
        this.additionalOrders = additionalOrders;
    }
}

export class MeetingBlock {
    start: DateTime;
    end: DateTime;
    topic: string;
    location: string;

    constructor(start: DateTime, end: DateTime, topic: string, location: string) {
        this.start = start
        this.end = end
        this.topic = topic
        this.location = location
    }
}

File: pkg/capa5/meeting.go
package capa5

import (
	"time"

	"github.com/derhabicht/cap-tools/pkg/members"
)

type Meeting struct {
	location         string
	address          string
	start            time.Time
	end              time.Time
	meetingType      MT
	topic            string
	blocks           []MeetingBlock
	uod              []members.Uniform
	additionalOrders string
}

func NewMeeting(
	location string,
	address string,
	start time.Time,
	end time.Time,
	meetingType MT,
	topic string,
	blocks []MeetingBlock,
	uod []members.Uniform,
	additionalOrders string,
) Meeting {
	return Meeting{
		location:         location,
		address:          address,
		start:            start,
		end:              end,
		meetingType:      meetingType,
		topic:            topic,
		blocks:           blocks,
		uod:              uod,
		additionalOrders: additionalOrders,
	}
}
